lib/ansible/modules/crypto/openssh_cert.py

```diff
@@ -0,0 +1,542 @@
+
+        if _check_state():
+            proc = module.run_command([self.ssh_keygen, '-L', '-f', self.path], environ_update=dict(TZ="UTC"))
+            self.cert_info = proc[1].split()
+            principals = re.findall("(?<=Principals:)(.*)(?=Critical)", proc[1], re.S)[0].split()
+            principals = list(map(str.strip, principals))
+            cert_type = re.findall("( user | host )", proc[1])[0].strip()
+            validity = re.findall("(from (\\d{4}-\\d{2}-\\d{2}T\\d{2}(:\\d{2}){2}) to (\\d{4}-\\d{2}-\\d{2}T\\d{2}(:\\d{2}){2}))", proc[1])
+            if validity:
+                cert_valid_from = self.convertToDatetime(module, validity[0][1])
+                cert_valid_to = self.convertToDatetime(module, validity[0][3])
+            else:
+                cert_valid_from = datetime(MINYEAR, 1, 1)
+                cert_valid_to = datetime(MAXYEAR, 12, 31)
+        else:
+            return False
+
+        def _check_perms(module):
+            file_args = module.load_file_common_arguments(module.params)
+            return not module.set_fs_attributes_if_different(file_args, False)
+
+        def _check_type():
+            return self.type == cert_type
+
+        def _check_principals():
+            return self.principals == principals
+
+        def _check_validity(module):
+            if self.valid_from == "always":
+                earliest_time = datetime(MINYEAR, 1, 1)
+            elif self.isRelative(self.valid_from):
+                earliest_time = None
+            else:
+                earliest_time = self.convertToDatetime(module, self.valid_from)
+
+            if self.valid_to == "forever":
+                last_time = datetime(MAXYEAR, 12, 31)
+            elif self.isRelative(self.valid_to):
+                last_time = None
+            else:
+                last_time = self.convertToDatetime(module, self.valid_to)
+
+            if earliest_time:
+                if not (earliest_time - cert_valid_from).total_seconds == 0.0:
+                    return False
+            if last_time:
+                if not (last_time - cert_valid_to).total_seconds == 0.0:
+                    return False
+
+            if self.valid_at:
+                if cert_valid_from <= self.convertToDatetime(module, self.vaid_at) <= cert_valid_to:
+                    return True
+
+            if earliest_time and last_time:
+                return True
+
+            return False
+
+        if not perms_required:
+            return _check_type() and _check_principals() and _check_validity(module)
+
+        return _check_perms(module) and _check_type() and _check_principals() and _check_validity(module)
+
+    def dump(self):
+
+        """Serialize the object into a dictionary."""
+
+        def filterKeywords(arr, keywords):
+            concated = []
+            string = ""
+            for word in arr:
+                if word in keywords:
+                    concated.extend([string])
```

Comment:
Why don't you do `concated.append(string)`? That's more efficient and easier to read.